src/tools/perf/lib/libperf.c

```diff
@@ -1267,6 +1282,163 @@ static ucs_status_t ucp_perf_test_send_local_data(ucx_perf_context_t *perf,
     return status;
 }
 
+static void err_cb(void *arg, ucp_ep_h ep, ucs_status_t status)
+{
+    ucs_diag("error handling callback was invoked with status %d (%s)", status,
+             ucs_status_string(status));
+}
+
+/*
+ * Pack array length and array contents into a buffer.
+ *
+ * @param [out] ptr    Pointer to the buffer.
+ * @param [in]  length Length of the array.
+ * @param [in]  value  Pointer to the array.
+ */
+static void ucp_perf_pack_array(void **ptr, uint64_t length, const void *value)
+{
+    *ucs_serialize_next(ptr, uint64_t) = length;
+    if (length > 0) {
+        memcpy(ucs_serialize_next_raw(ptr, void, length), value, length);
+    }
+}
+
+static size_t ucp_perf_pack_daemon_info(const ucx_perf_context_t *perf,
+                                        void *buffer, size_t buffer_size,
+                                        const struct sockaddr_storage *address)
+{
+    void *p               = buffer;
+    size_t address_length = (address != NULL) ? sizeof(*address) : 0;
```

Comment:
maybe pass the length? the actual ipv4/ipv6 length is not really as big as sockaddr_storage